librz/asm/arch/6502/6502dis.c

```diff
@@ -143,52 +143,42 @@ static int _6502Disass(ut64 pc, RzAsmOp *op, const ut8 *buf, ut64 len) {
 	int i;
 	for (i = 0; ops[i].name != NULL; i++) {
 		if (ops[i].op == buf[0]) {
-			char *buf_asm = rz_str_newf("invalid");
 			int len = ops[i].len;
 			switch (ops[i].len) {
 			case 1:
-				free(buf_asm);
-				buf_asm = rz_str_newf("%s", ops[i].name);
+				rz_asm_op_setf_asm(op, "%s", ops[i].name);
 				break;
 			case 2:
 				if (len > 1) {
-					free(buf_asm);
-					buf_asm = rz_str_newf(ops[i].name, buf[1]);
+					rz_asm_op_setf_asm(op, ops[i].name, buf[1]);
 				} else {
-					free(buf_asm);
-					buf_asm = rz_str_newf("truncated");
+					rz_asm_op_set_asm(op, "truncated");
 					len = -1;
 				}
 				break;
 			case 3:
 				if (len > 2) {
-					free(buf_asm);
-					buf_asm = rz_str_newf(ops[i].name, buf[1] + 0x100 * buf[2]);
+					rz_asm_op_setf_asm(op, ops[i].name, buf[1] + 0x100 * buf[2]);
 				} else {
-					free(buf_asm);
-					buf_asm = rz_str_newf("truncated");
+					rz_asm_op_set_asm(op, "truncated");
 					len = -1;
 				}
 				break;
 			case 4:
 				if (len > 3) {
-					free(buf_asm);
-					buf_asm = rz_str_newf(ops[i].name, buf[1] + 0x100 * buf[2] + 0x10000 * buf[3]);
+					rz_asm_op_setf_asm(op, ops[i].name, buf[1] + 0x100 * buf[2] + 0x10000 * buf[3]);
 				} else {
-					free(buf_asm);
-					buf_asm = rz_str_newf("truncated");
+					rz_asm_op_set_asm(op, "truncated");
 					len = -1;
 				}
 				break;
 			default:
-				free(buf_asm);
 				goto beach;
 			}
-			rz_strbuf_set(&op->buf_asm, buf_asm);
-			free(buf_asm);
 			return len;
 		}
 	}
 beach:
+	rz_asm_op_set_asm(op, "invalid");
```

Comment:
no. this overwrites the other `rz_asm_op_set_asm`. move this into the `default` case. (see the other comment.)